Urban-Fun/View Controllers/ActivityDetailsViewController.m

```diff
@@ -6,18 +6,90 @@
 //
 
 #import "ActivityDetailsViewController.h"
+#import "PFUser.h"
+#import "HelperClass.h"
 
 @interface ActivityDetailsViewController ()
 @property (strong, nonatomic) IBOutlet UILabel *detailsTitle;
+@property (strong, nonatomic) IBOutlet UILabel *detailsHostLabel;
+@property (strong, nonatomic) IBOutlet UILabel *detailsLocationLabel;
+@property (strong, nonatomic) IBOutlet UIButton *detailsFavoriteButton;
+@property (strong, nonatomic) IBOutlet UIButton *detailsJoinButton;
+@property (strong, nonatomic) IBOutlet UILabel *detailsAttendanceLabel;
+@property (strong, nonatomic) IBOutlet UILabel *detailsDescriptionLabel;
+@property (strong,nonatomic) PFUser *currentUser;
 
+- (void) setUpView;
 @end
 
 @implementation ActivityDetailsViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
     // Do any additional setup after loading the view.
+    [self setUpView];
+}
+
+- (IBAction)didTapJoin:(id)sender {
+    PFUser *currentUser = [PFUser currentUser];
+    if ([self.activity.host.objectId isEqualToString:currentUser.objectId]){
+        }
+    else{
+        [Activity updateAttendanceListWithUserId:currentUser.objectId withActivity:self.activity withCompletion:^(BOOL succeeded, NSError * _Nullable error) {
+            if ([self.activity.attendanceList containsObject:currentUser.objectId]){
+                [self.detailsJoinButton setSelected:YES];
+            }
+            else if (![self.activity.attendanceList containsObject:currentUser.objectId]){
+                [self.detailsJoinButton setSelected:NO];
+            }
+            [self setUpView];
+            
+        }];
+        
+    }
+    
+}
+
+- (void)setUpView {
     self.detailsTitle.text = self.activity.title;
+    self.detailsHostLabel.text = [NSString stringWithFormat:@"%@%@", @"Host - ", self.activity.host[@"username"]];
+    self.detailsAttendanceLabel.text = [NSString stringWithFormat:@"%@%lu", @"Attendance - ", (unsigned long)self.activity.attendanceList.count];
```

Comment:
Ok, I see. When I was creating this string, I had originally used "%@" instead of "%lu" and Xcode threw an error...when I pressed "fix", Xcode made the change to "%lu" and put the "(unsigned long)" there.